File: packages/emitter/testcases/common/expected/types/expected-type-threading/NullishFull.cs
namespace TestCases.common.types.expectedtypethreading
{
    [global::Tsonic.Internal.ModuleContainerAttribute]
    public static class NullishFull
    {
        public static int basicNullish(int? value)
        {
            return value ?? (int)100;
        }

        public static int nestedNullish(int? a, int? b)
        {
            return a ?? (int?)b ?? (int)999;
        }

        public static int nullishWithExpr(int? value, int fallback)
        {
            return value ?? fallback;
        }

        public static int nullishInVar(int? value)
        {
            int result = value ?? (int)42;
            return result;
        }

        public static int nullishInIf(int? value, bool condition)
        {
            if (condition)
            {
                return value ?? (int)50;
            }
            return value ?? (int)60;
        }
    }
}

File: packages/emitter/testcases/common/types/expected-type-threading/NullishFull.ts
import { int } from "@tsonic/core/types.js";

// ?? with nullable parameter - basic case
export function basicNullish(value: int | null): int {
  return value ?? 100;
}

// Nested ?? chain
export function nestedNullish(a: int | null, b: int | null): int {
  return a ?? b ?? 999;
}

// ?? with expression RHS
export function nullishWithExpr(value: int | null, fallback: int): int {
  return value ?? fallback;
}

// ?? in variable declaration
export function nullishInVar(value: int | null): int {
  const result: int = value ?? 42;
  return result;
}

// ?? in return inside if
export function nullishInIf(value: int | null, condition: boolean): int {
  if (condition) {
    return value ?? 50;
  }
  return value ?? 60;
}
